util.py

```diff
@@ -11,41 +11,28 @@
 log = logging.getLogger(__name__)
 
 
-async def prompt_for_turn(
+async def menu(
 	bot: commands.Bot,
 	user: discord.User,
-	battlecontext: BattleContext,
+	items: list[str],
 	use_channel: Optional[discord.TextChannel] = None
-) -> Turn:
-	"""Prompt the given user for a turn.
-
-	:param bot: The discord bot
-	:param user: The discord user
-	:param battlecontext: The :class:`BattleContext` that will be given to the user.
-	:param use_channel: Override the channel that is used to communicate with the user. By default, it will use the user's :class:`discord.DMChannel` to send the prompt. Primarily used for tests.
-	:returns: The turn that the user made.
-	TODO: prompt user for what type of turn
-	"""
+):
+	"""Create menu for the user to choose between several options."""
 
 	if use_channel:
 		channel = use_channel
 	else:
 		if not user.dm_channel:
 			await user.create_dm()
 		channel = user.dm_channel
-	embed = discord.Embed(
-		title=battlecontext.pokemon.Name,
-		description=
-		f"{battlecontext.pokemon.CurrentHP} HP {taggify(type_to_string(battlecontext.pokemon.Type))} {taggify(status_to_string(battlecontext.pokemon.StatusEffects))}"
-	)
-	for i, move in enumerate(battlecontext.pokemon.Moves):
-		embed.add_field(
-			name=f"{RESPONSE_REACTIONS[i]}: {move['Name']}",
-			value=
-			f"{taggify(type_to_string(move['Type']))} {move['CurrentPP']}/{move['MaxPP']}",
-			inline=False
-		)
-	msg: Message = await channel.send(content="Select a move", embed=embed)
+
+	embed = discord.Embed(title=items[0], description=items[1])
+	rest_of_list = items[2:]
+
+	for item in rest_of_list:
+		embed.add_field(name=rest_of_list[0], value=rest_of_list[1])
```

Comment:
This displays the same first 2 items from the list (index 0, and index 1) for each item that is in the list?

This would probably work better if `items` was `list[tuple[str]]`

Reply:
I knew this was a big issue but I left it as that since I couldn't think of any other way that would work. I'll try to see how having it as `list[tuple[str]]` would change it.